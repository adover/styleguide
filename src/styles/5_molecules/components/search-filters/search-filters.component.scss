@import '../../../1_settings/_config';
@import '../../../4_atoms/heading';
@import '../../../4_atoms/list';
@import '../../../4_atoms/button';
@import '../../../5_molecules/accordion';
@import '../../../2_tools/mixins/media-query';
@import '../../../2_tools/mixins/apply-spacing';

:host {
	display: block;

	@include media-query($size: mobile) {
		@include section-margin($size: default);
	}

	h3 {
		@include heading-underlined();
		@include accordion-icon();

		cursor: pointer;
		padding-right: $spacing-12;
		position: relative;

		@include media-query($size: mobile, $max: true) {
			@include accordion-item-header();
		}

		&.open {
			@include accordion-icon($state: open);
		}
	}

	.filtersContent {
		overflow: hidden;
	}

	::ng-deep {
		ul {
			@include media-query($size: mobile, $max: true) {
				@include accordion-init();
			}

			@include list-init($reset: true);

			li {
				font-size: map-deep-get($config, sizes, paragraph, small);

				@include media-query($size: mobile, $max: true) {
					&:not(:last-child) {
						margin-bottom: 0;
						padding-bottom: 0;
					}
				}

				button {
					@include button-link();

					white-space: normal;

					&:hover {
						color: $link-color;
						text-decoration: underline;
					}
				}

				ul li {
					margin: 0;
					padding: 0;

					&:empty {
						display: none;
					}
				}
			}
		}
	}
}
