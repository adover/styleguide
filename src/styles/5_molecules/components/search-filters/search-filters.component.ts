import { Component, Input, OnInit } from '@angular/core';
import { openClose } from '../../../animations';
import { BreakPointService, Device } from '../../../services/break-point.service';

@Component({
	selector: 'cdx-search-filters',
	template: `
		<h3 class="heading--4" [class.open]="isOpen" (click)="toggleFilters()">
			{{ title }}
		</h3>
		<div [@openClose]="isOpen" class="filtersContent">
			<ng-content></ng-content>
		</div>
	`,
	styleUrls: ['./search-filters.component.scss'],
	animations: [...openClose],
})
export class SearchFiltersComponent implements OnInit {
	@Input() isOpen = true;
	@Input() title = 'Categories';

	constructor(public breakPointService: BreakPointService) {}

	ngOnInit(): void {
		this.breakPointService.device$.subscribe((device) => {
			this.isOpen = device !== Device.MOBILE;
		});
	}

	toggleFilters = () => {
		this.isOpen = !this.isOpen;
	};
}
