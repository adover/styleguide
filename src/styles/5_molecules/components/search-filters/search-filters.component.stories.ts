import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { storiesOf } from '@storybook/angular';
import { SearchFiltersComponent } from './search-filters.component';
import { THEME_CONSTANTS } from '../../../1_settings/theme';
import { StyleGuideConfig } from '../../../styleguide.config';
import { APP_SETTINGS_PROVIDER, ApiService } from '../../../services';
import { HttpClientTestingModule } from '@angular/common/http/testing';

storiesOf('Molecules| SearchFilters', module).add('default', () => {
	const filterItems = [
		'Bakery (2)',
		'Seafood (92)',
		'Baby Care(10)',
		'Baking & Cooking (103)',
		'Drinks- Hot & Cold(29)',
	];
	return {
		moduleMetadata: {
			declarations: [SearchFiltersComponent],
			imports: [BrowserAnimationsModule, HttpClientTestingModule],
			providers: [
				APP_SETTINGS_PROVIDER,
				ApiService,
				{
					provide: StyleGuideConfig,
					useValue: {
						breakpoints: THEME_CONSTANTS.breakpoints,
					},
				},
			],
		},
		template: `
        <cdx-search-filters>
          <ul>
            <li
              *ngFor="let item of filterItems"
            >
              <button
              >
                {{ item | titlecase }} (2)
              </button>
            </li>
          </ul>
        </cdx-search-filters>
    `,
		props: {
			filterItems: filterItems,
		},
	};
});
