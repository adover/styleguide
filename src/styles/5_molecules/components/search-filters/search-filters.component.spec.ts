import { HttpClientTestingModule } from '@angular/common/http/testing';
import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { SearchFiltersComponent } from './search-filters.component';
import { ApiService, APP_SETTINGS_PROVIDER, AppSettingsService, BreakPointService } from '../../../services';
import { StyleGuideConfig } from '../../../styleguide.config';
@Component({
	template: `
		<cdx-search-filters [isOpen]="isOpen">
			<div *ngFor="let item of filterItems">{{ item }}</div>
		</cdx-search-filters>
	`,
})
class HostComponent {
	filterItems = ['Bakery (2)', 'Seafood (92)', 'Baby Care(10)', 'Baking & Cooking (103)', 'Drinks- Hot & Cold(29)'];
	isOpen = true;
}

describe('Search Filters Component', () => {
	let hostComponent: HostComponent;
	let hostFixture: ComponentFixture<HostComponent>;
	let component: SearchFiltersComponent;
	let fixture: ComponentFixture<SearchFiltersComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [SearchFiltersComponent, HostComponent],
			providers: [
				APP_SETTINGS_PROVIDER,
				{
					provide: AppSettingsService,
					useValue: {
						getEndpoint: () => '/api',
						getTransition: () => '',
						getBreakpoints: () => ({
							mobile: '',
							desktop: '',
							tablet: '',
						}),
					},
				},
				ApiService,
				BreakPointService,
				{
					provide: StyleGuideConfig,
					useValue: {
						breakpoints: {
							mobile: '640px',
							tablet: '970px',
							desktop: '1200px',
						},
					},
				},
			],
			imports: [BrowserAnimationsModule, RouterTestingModule, HttpClientTestingModule],
		}).compileComponents();
	}));

	beforeEach(() => {
		hostFixture = TestBed.createComponent(HostComponent);
		hostComponent = hostFixture.componentInstance;
		hostFixture.detectChanges();

		fixture = TestBed.createComponent(SearchFiltersComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(hostComponent).toBeTruthy();
	});

	it('render filters as open', () => {
		expect(hostFixture).toMatchSnapshot();
	});

	it('render filters as closed', () => {
		hostComponent.isOpen = false;
		hostFixture.detectChanges();
		expect(hostFixture).toMatchSnapshot();
	});

	it('toggleFilters shoudl toggle the open state', () => {
		component.toggleFilters();
		expect(component.isOpen).toBeTruthy();
	});

	it('clicking on heading should toggle the open state', () => {
		component.toggleFilters = jest.fn();
		fixture.detectChanges();
		fixture.debugElement.query(By.css('h3')).nativeElement.click();
		expect(component.toggleFilters).toHaveBeenCalled();
	});
});
