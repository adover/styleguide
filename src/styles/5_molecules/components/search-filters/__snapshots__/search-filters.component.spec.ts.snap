// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Filters Component render filters as closed 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  filterItems={[Function Array]}
  isOpen="false"
>
  <cdx-search-filters
    class="ng-tns-c29-0"
  >
    <h3
      class="heading--4 ng-tns-c29-0"
    >
       Categories 
    </h3>
    <div
      class="filtersContent ng-tns-c29-0 ng-trigger ng-trigger-openClose"
      style="height:0px;0:height;"
    >
      <div
        class="ng-star-inserted"
        style=""
      >
        Bakery (2)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Seafood (92)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Baby Care(10)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Baking & Cooking (103)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Drinks- Hot & Cold(29)
      </div>
      
    </div>
  </cdx-search-filters>
</ng-component>
`;

exports[`Search Filters Component render filters as open 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  filterItems={[Function Array]}
  isOpen={[Function Boolean]}
>
  <cdx-search-filters
    class="ng-tns-c29-0"
  >
    <h3
      class="heading--4 ng-tns-c29-0"
    >
       Categories 
    </h3>
    <div
      class="filtersContent ng-tns-c29-0 ng-trigger ng-trigger-openClose"
      style="height:0px;0:height;"
    >
      <div
        class="ng-star-inserted"
        style=""
      >
        Bakery (2)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Seafood (92)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Baby Care(10)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Baking & Cooking (103)
      </div>
      <div
        class="ng-star-inserted"
        style=""
      >
        Drinks- Hot & Cold(29)
      </div>
      
    </div>
  </cdx-search-filters>
</ng-component>
`;
